### PR TITLE
Sync (3/N): Modules

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/1174433894299346/1204078441653387

### Description
This PR moves all the Saved Sites related changes to it's own modules

### CODE REVIEW

**Comment by malmstein** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2812** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2812" target="_blank">Graphite</a> 
    * **PR #2821** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2821" target="_blank">Graphite</a> 
      * **PR #2836** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2836" target="_blank">Graphite</a> 
        * **PR #2856** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2856" target="_blank">Graphite</a> 
          * **PR #2889** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2889" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2889?utm_source=stack-comment).

File: saved-sites/saved-sites-store/build.gradle [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    implementation Square.okio
+    implementation AndroidX.room.runtime
+    implementation AndroidX.room.rxJava2
+    implementation AndroidX.room.ktx
+    implementation KotlinX.coroutines.core
+    implementation KotlinX.coroutines.android
+
+    kapt AndroidX.room.compiler
+
+    testImplementation Testing.junit4
```
Comment (cmonfortep) [inline saved-sites/saved-sites-store/build.gradle:49]: not sure we need all the test dependencies if the module has not tests.

File: saved-sites/saved-sites-store/build.gradle [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    id 'kotlin-kapt'
+}
+
+apply from: "$rootProject.projectDir/gradle/android-library.gradle"
+
+android {
+    lint {
+        baseline file("lint-baseline.xml")
+    }
+    testOptions {
```
Comment (cmonfortep) [inline saved-sites/saved-sites-store/build.gradle:29]: probably we don't need this.

File: saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/SavedSitesRepository.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import com.duckduckgo.savedsites.api.models.BookmarkFolderItem
+import com.duckduckgo.savedsites.api.models.FolderBranch
+import com.duckduckgo.savedsites.api.models.SavedSite
+import com.duckduckgo.savedsites.api.models.SavedSite.Bookmark
+import com.duckduckgo.savedsites.api.models.SavedSite.Favorite
+import com.duckduckgo.savedsites.api.models.SavedSites
+import io.reactivex.Single
+import kotlinx.coroutines.flow.Flow
+
+interface SavedSitesRepository {
```
Comment (cmonfortep) [inline saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/SavedSitesRepository.kt:29]: We need to add docs to api interfaces.

File: saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/service/SavedSitesExporter.kt [hunk 4]
```diff
@@ -14,14 +14,16 @@
...
+import android.net.Uri
 
-data class BookmarkFolder(
-    val id: String,
-    val name: String,
-    val parentId: String,
-    val numBookmarks: Int = 0,
-    val numFolders: Int = 0,
-) : Serializable
+interface SavedSitesExporter {
```
Comment (cmonfortep) [inline saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/service/SavedSitesExporter.kt:21]: We need to add docs to api interfaces.

File: saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/service/SavedSitesImporter.kt [hunk 5]
```diff
@@ -14,10 +14,16 @@
...
 
-data class BookmarkFolderItem(
-    val depth: Int,
-    val bookmarkFolder: BookmarkFolder,
-    val isSelected: Boolean = false,
-)
+import android.net.Uri
+import com.duckduckgo.savedsites.api.models.SavedSite
+
+interface SavedSitesImporter {
```
Comment (cmonfortep) [inline saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/service/SavedSitesImporter.kt:22]: ditto.

File: saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/service/SavedSitesManager.kt [hunk 6]
```diff
@@ -14,11 +14,11 @@
...
+package com.duckduckgo.savedsites.api.service
 
-import com.duckduckgo.sync.store.Entity
+import android.net.Uri
 
-data class BookmarkFolderBranch(
-    val bookmarks: List<Entity>,
-    val folders: List<Entity>,
-)
+interface SavedSitesManager {
```
Comment (cmonfortep) [inline saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/service/SavedSitesManager.kt:21]: ditto.

File: saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/models/TreeNode.kt [hunk 7]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.duckduckgo.app.bookmarks.model
+package com.duckduckgo.savedsites.api.models
 
 class TreeNode<T>(val value: T) {
```
Comment (cmonfortep) [inline saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/models/TreeNode.kt:19]: is this class used in :api?

File: saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/SavedSitesRepository.kt [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+    fun getBookmarksObservable(): Single<List<Bookmark>>
+
+    fun getBookmark(url: String): Bookmark?
+
+    fun getFavorites(): Flow<List<Favorite>>
+
+    fun getFavoritesObservable(): Single<List<Favorite>>
+
+    fun getFavoritesSync(): List<Favorite>
```
Comment (cmonfortep) [inline saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/SavedSitesRepository.kt:56]: is this something we forgot to remove? not sure we want to see references to sync inside this module.

File: saved-sites/saved-sites-store/src/main/java/com/duckduckgo/savedsites/store/SyncEntitiesDao.kt [hunk 9]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.duckduckgo.sync.store
+package com.duckduckgo.savedsites.store
```
Comment (cmonfortep) [inline saved-sites/saved-sites-store/src/main/java/com/duckduckgo/savedsites/store/SyncEntitiesDao.kt:17]: We should update interface names to remove `Sync`

File: saved-sites/saved-sites-store/src/main/java/com/duckduckgo/savedsites/store/SyncRelationsDao.kt [hunk 10]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.duckduckgo.sync.store
+package com.duckduckgo.savedsites.store
```
Comment (cmonfortep) [inline saved-sites/saved-sites-store/src/main/java/com/duckduckgo/savedsites/store/SyncRelationsDao.kt:17]: ditto (sync reference)

File: saved-sites/saved-sites-api/build.gradle [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+apply from: "$rootProject.projectDir/gradle/android-library.gradle"
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+dependencies {
```
Comment (cmonfortep) [inline saved-sites/saved-sites-api/build.gradle:30]: can you review dependencies here? :api module should be pretty flat, e.g: we shouldn't have a dependency with :store.

**Review by cmonfortep — COMMENTED** [review]
Took a pass and left some comments.

**Pull request merged**